modules/social_features/social_event/modules/social_event_managers/src/SocialEventManagersAccessHelper.php

```diff
@@ -63,4 +64,42 @@ public static function getEntityAccessResult(NodeInterface $node, $op, AccountIn
     return AccessResult::neutral();
   }
 
+  /**
+   * Event enrollment acces for given operation.
+   */
+  public function eventEnrollmentAccessCheck(EventEnrollment $event_enrollment, string $op, AccountInterface $account): AccessResult {
+    // This allows view access to event_enrollment entities for users which are
+    // the recipients of the event enrollment but not the owner of the entity.
+    // For example a site manager can create an enrollment for a specific user.
+    if ($op !== 'view') {
+      // If we are doing different operations than viewing, then let other
+      // access checks to determine the access.
+      return AccessResult::neutral();
+    }
+
+    $enrollment_status = $event_enrollment->get('field_enrollment_status')->getString();
+    if (!(bool) $enrollment_status) {
+      // Return neutral and let other access checks acts on this.
+      return AccessResult::neutral();
+    }
+
+    $owner = $event_enrollment->getOwner();
+    if (!$owner->hasPermission('view published event enrollment entities')) {
+      // Return neutral and let other access checks acts on this.
+      return AccessResult::neutral();
+    }
+
+    $field_account_id = $event_enrollment->get('field_account')->getString();
+    if ((int) $field_account_id !== (int) $account->id()) {
+      // If current user is not the recipient then let other
+      // part of the system determine the access, by default it should
+      // be access denied, but it could be overwritten, so we don't make
+      // any assumptions here.
+      return AccessResult::neutral();
+    }
+
+    // The user is a recipient, so we allow access to event enrollment.
+    return AccessResult::allowed();
```

Comment:
@nkoporec it works, but wondering if there is something we can rewrite for this? In general I get a bit scared when I see an access check that end with allowed() because if something would change in the flow in the future it could possible create a security issue.

If it would be allowedIf() I would already feel a bit better.

Reply:
hmm I get your point, but I don't think that there is a way to rewrite this without having the allowed() since it needs to override the default behavior. We could add the `allowedIf((int) $field_account_id === (int) $account->id())` and get rid of the if before that , if that works better for you ?

Reply:
Yeah, I think that sounds like a good plan, if you can do that I will approve and merge.